src/index.js

```diff
@@ -1,4 +1,89 @@
-// $(document).ready(function(){
+$(document).ready(function(){
+//selecting html elements for use in color change functions
+document.getElementById("name").onkeydown = function() {nameInputBackgroundChange()};
+document.getElementById("mail").onkeydown = function() {mailInputBackgroundChange()};
+document.getElementById("subject").onkeydown = function() {subjectInputBackgroundChange()};
+document.getElementById("msg").onkeydown = function() {msgInputBackgroundChange()};
+
+//selecting html elements for use in reverting colors back to original color upon focus change
+document.getElementById("name").onchange = function() {nameInputChange()};
+document.getElementById("mail").onchange = function() {mailInputChange()};
+document.getElementById("subject").onchange = function() {subjectInputChange()};
+document.getElementById("msg").onchange = function() {msgInputChange()};
+
+// functions to change input/text area color and background color based on userinput
+function nameInputBackgroundChange() {
+  var nameField = document.getElementById("name");
+  nameField.style.backgroundColor = "white";
+  nameField.style.color = "#4D766E";
+}
+
+function nameInputChange() {
```

Comment:
## Here is another example of non-DRY code.

### The Problem:

You've declared two functions to handle the same event (though, you are handling them as separate events...a `keydown` and a `change`).

### Listening for a single event:

It's simpler to think of this as if there is only one event happening. Let's choose to think simpler - either:

- The user is pressing a key on the keyboard.
- The value of the input field is changing.

**Therefore, we can listen for a single event.**

`keydown`, `keyup`, and `keypress` events are slightly easier to handle in jQuery, so for the sake of MVP, let's  choose to consider this event to be one of these keyboard events.

In jQuery, we can listen for and handle an event with the `.on` method. For instance, in this case you'd do:

```
$nameInput.on( _____, ______ );
```

**Read the documentation on jQuery's official docs page to determine what the two inputs should be!** 

### Handling a Single Event, with Multiple Conditions:

Though *there are not **two separate events** occurring, there are **two separate conditions** to consider:*

1. After the user presses a key, the value in the input field contains some text.
2. After the user presses a key, the value in the input field does not contain any text.

So, why not use *a single event handler function* with *a conditional statement inside*?

```  
function handleNameInputChange() {
  // A console.log here of the value in the input field will help you debug when you're listening for the wrong event
  console.log('FILL_ME_IN');

  // If the input field is empty
  if ('FILL_ME_IN') {
    // Change `background-color` to `'#4D766E'`, and `color` to `'white'`

  } else {
    // Change `background-color` to `'white'`, and `color` to `'#4D766E'` 

  }
}
```

**I'll leave the jQuery code as an exercise for you.**